apps/studio/src/hooks/useResourceLocalViewHistory.ts

```diff
@@ -0,0 +1,50 @@
+const LOCAL_VIEW_HISTORY_KEY = "localViewHistory"
+
+interface LocalViewHistory {
+  resourceId: string
+  dateTime: Date
+}
+
+export const useResourceLocalViewHistory = ({ siteId }: { siteId: string }) => {
+  const storageKeyForSiteId = `${LOCAL_VIEW_HISTORY_KEY}-${siteId}`
+
+  const get = (): LocalViewHistory[] => {
+    let history: LocalViewHistory[] = []
+    const storedHistory = localStorage.getItem(storageKeyForSiteId)
+
+    if (storedHistory) {
+      try {
+        history = JSON.parse(storedHistory) as LocalViewHistory[]
+      } catch (error) {
+        console.error("Failed to parse local view history from storage:", error)
+      }
+    }
+    return history
+  }
+
+  const upsert = ({ resourceId }: { resourceId: string }): void => {
+    const localViewHistory: LocalViewHistory[] = get()
+
+    const existingEntryIndex: number = localViewHistory.findIndex(
+      (entry) => entry.resourceId === resourceId,
+    )
+
+    const entryToUpdate: LocalViewHistory | undefined =
+      localViewHistory[existingEntryIndex]
+
+    if (entryToUpdate) {
+      // Remove the existing entry
+      localViewHistory.splice(existingEntryIndex, 1)
+    }
+    // Add the new entry to the beginning of the array
+    localViewHistory.unshift({ resourceId, dateTime: new Date() })
```

Comment:
unshift is potentially expensive, but not really a concern for small number of items. a better way would be to use push and store in reverse order. Then can map in reverse index

Reply:
noted, thanks for sharing 😄 

the current approach `splice` it to only have 10 items (despite us only showing 3) so i wouldn't proceed to make this change. Furthermore, we prob wouldn't want to store uncapped number of items in localstorage too

Reply:
yea don't need to change, more fyi esp is only 10 items